### PR TITLE
feat: create custom url that redirects to the most recent snapshot

### LINKED ISSUE #45: Expose a stable link to the latest calibration network snapshot
**Issue summary**

We are storing calibration network snapshots in the Digital Ocean spaces and using them for Forest purposes which is good. We can do even better for the Filecoin community.

There is no publicly known link to the latest snapshot we host. We should find a way to expose it (latest by creation time) and afterwards talk with relevant folks to add it to the Filecoin documentation https://docs.filecoin.io/developers/infrastructure/networks/#calibration

Acceptance criteria:
- [ ] Calibration network snapshot is exposed with a stable link, e.g., `forest-calibration.chainsafe.io/snapshot/latest.car`. This should redirect to the newest snapshot in the Forest DO Spaces.
- [ ] Using the link directly should also work (at least as long as the snapshots are not pruned)
- [ ] Infrastructure/documentation Filecoin team is contacted, and the link is added.

**Other information and links**

### PR DESCRIPTION
**Pull request by lemmih**
**Summary of changes**
Changes introduced in this pull request:
- Added DigitalOcean Function that redirects to the most recent snapshot

**Reference issue to close (if applicable)**

Closes #45 

**Other information and links**

### CODE REVIEW

File: snapshot_function/packages/latest-snapshot/link/link.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    let compressed = !(args.compressed === 'false' || args.compressed === 'no' || args.compressed === '0');
+
+    // default to the calibnet network unless we're told otherwise
+    let network = args.network || 'calibnet';
+
+    const response = await fetch(base_url);
+    const body = await response.text();
+
+    const parser = new XMLParser();
+    let jObj = parser.parse(body);
```
Comment (LesnyRumcajs) [inline snapshot_function/packages/latest-snapshot/link/link.js:22]: what is `jObj`? Could we use a more explicit name?

**Review by LesnyRumcajs — CHANGES_REQUESTED** [review]

File: snapshot_function/packages/latest-snapshot/link/link.js
[the remark this answers is not in the document]
Comment (lemmih) [reply snapshot_function/packages/latest-snapshot/link/link.js:1]: Indeed. Will do.

File: snapshot_function/packages/latest-snapshot/link/link.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
+const fetch = require('node-fetch');
+const { XMLParser, XMLBuilder, XMLValidator } = require("fast-xml-parser");
+
+const region = 'fra1'
+const bucket = 'forest-snapshots'
+
+const base_url = 'https://' + region + '.digitaloceanspaces.com/' + bucket;
```
[the remark this answers is not in the document]
Comment (lemmih) [reply snapshot_function/packages/latest-snapshot/link/link.js:7]: Hmm, I don't actually know how to do this across all the services.

File: snapshot_function/packages/latest-snapshot/link/link.js (hunk 1 above)
Comment (lemmih) [reply snapshot_function/packages/latest-snapshot/link/link.js]: Renamed to `s3_listing`.

File: snapshot_function/packages/latest-snapshot/link/link.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (LesnyRumcajs) [reply snapshot_function/packages/latest-snapshot/link/link.js:7]: Let's not make it a blocker. Creating a separate issue for this is fine for me.

[the remark this answers is not in the document]
Comment (lemmih) [reply snapshot_function/packages/latest-snapshot/link/link.js:7]: #83

**Review by LesnyRumcajs — APPROVED** [review]

**Pull request merged**